### PR TITLE
Rename libsbn -> bito.

### LINKED ISSUE #353: Rename libsbn to bito!

### PR DESCRIPTION
**Pull request by davidrich27**
## Description

This PR resolves the issue of changing the project name from `libsbn` to `bito`. 
- All references to libsbn were replaced with bito (source code, pybind library).
- A short blurb was added to the README about name origins, as well as old library name.

Closes #353 

## Tests

No testing necessary.  

## Checklist:

* [x] Code follows our detailed [contribution guidelines](CONTRIBUTING.md)
* [x] `clang-format` has been run
* [x] TODOs have been eliminated from the code
* [x] Comments are up to date, document intent, and there are no commented-out code blocks

### CODE REVIEW

**Comment by davidrich27** [comment]
- https://www.merriam-webster.com/dictionary/bito 
- https://en.wikipedia.org/wiki/Balanites_aegyptiaca

**Comment by davidrich27** [comment]
References to outside links using `libsbn` naming:
- README.md:
    - [Docker Repository on Quay](https://quay.io/repository/matsengrp/libsbn)
    - [Travis CI status](https://travis-ci.org/phylovi/libsbn)
- doc/tex/README.md: 
    - For API documentation see [phylovi.github.io/bito/](https://phylovi.github.io/bito/).

File: README.md [hunk 1]
```diff
@@ -1,9 +1,11 @@
-# libsbn
+# bito
 
-[![Docker Repository on Quay](https://quay.io/repository/matsengrp/libsbn/status "Docker Repository on Quay")](https://quay.io/repository/matsengrp/libsbn)
-[![Travis CI status](https://travis-ci.org/phylovi/libsbn.svg?branch=main)](https://travis-ci.org/phylovi/libsbn)
+[![Docker Repository on Quay](https://quay.io/repository/matsengrp/bito/status "Docker Repository on Quay")](https://quay.io/repository/matsengrp/bito)
```
Comment (matsen) [inline README.md:3]: We'll need to change these... if we want to continue using Quay then please make an account and send me your login info.

File: doc/index.rst [hunk 2]
```diff
@@ -1,8 +1,8 @@
-Welcome to libsbn's documentation!
+Welcome to bito's documentation!
```
Comment (matsen) [inline doc/index.rst:1]: Have you tried making the docs using `make docs`?

**Comment by 4ment** [comment]
There are some `libsbn` references in the repo: [CONTRIBUTING.md](../blob/353-rename-to-bito/CONTRIBUTING.md), [libsbn.rst](../blob/353-rename-to-bito/doc/modules/libsbn.rst), [libsbn.beagle_flags.rst](../blob/353-rename-to-bito/doc/modules/libsbn.beagle_flags.rst), [main.tex](../blob/353-rename-to-bito/doc/tex/main.tex). Should we change them now? We could add a short comment explaining that bito was formerly known as libsbn.

**Comment by davidrich27** [comment]
Thanks @4ment.  I added those missed references.  I think that should be all of them.

**Review by 4ment — APPROVED** [review]

**Comment by matsen** [comment]
@davidrich27 did you check that the `make docs` works properly?

**Comment by matsen** [comment]
Thank you, @davidrich27 !

**Pull request merged**